### PR TITLE
CODENVY-170 add promise-based authorization method to OAUth authenticators; prevent project synchronizing when factory used

### PR DESCRIPTION
**Pull request by mshaposhnik**
…ators; prevent project synchronizing when factory used

Signed-off-by: Max Shaposhnik mshaposhnik@codenvy.com

### CODE REVIEW

**Comment by mshaposhnik** [comment]
@vparfonov  @skabashnyuk

**Comment by vparfonov** [comment]
good job!

File: core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/oauth/OAuth2Authenticator.java [hunk 1]
```diff
@@ -21,5 +22,7 @@
 
     void authorize(String authenticatorUrl, AsyncCallback<OAuthStatus> callback);
 
+    Promise<OAuthStatus> authorize(String authenticationUrl);
```
Comment (sleshchenko) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/oauth/OAuth2Authenticator.java]: authorize but authenticationUrl
Mb authorizationUrl? or authenticate?

**Comment by skabashnyuk** [comment]
ок

File: core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/oauth/OAuth2Authenticator.java [hunk 2]
```diff
@@ -19,7 +20,9 @@
  */
 public interface OAuth2Authenticator {
 
-    void authorize(String authenticatorUrl, AsyncCallback<OAuthStatus> callback);
+    void authenticate(String authenticatorUrl, AsyncCallback<OAuthStatus> callback);
```
Comment (sleshchenko) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/oauth/OAuth2Authenticator.java]: first method with authenticatorUrl and second method with authenticationUrl. If there isn't any difference between them let's make them the same

**Comment by sleshchenko** [comment]
other OK

**Pull request merged**